tracer/test/Datadog.Trace.Tests/DatabaseMonitoring/DatabaseMonitoringPropagatorTests.cs

```diff
@@ -115,10 +117,55 @@ public void ExpectedCommentInjectedV1()
 
             traceParentInjectedValue.Should().Be(traceParentInjected);
             returnedComment.Should().Be(expectedComment);
         }
+
+        [Theory]
+        [InlineData("full", "sqlclient", SamplingPriorityValues.UserKeep, true, "01000000000000BEEF0000000000000000000000000000CAFE")]
+        [InlineData("full", "sqlclient", SamplingPriorityValues.UserReject, true, "00000000000000BEEF0000000000000000000000000000CAFE")]
+        [InlineData("nope", "sqlclient", SamplingPriorityValues.UserKeep, false, null)]
+        // disabled for all db types except mysql for now
+        [InlineData("full", "npgsql", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "sqlite", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "oracle", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "mysql", SamplingPriorityValues.UserKeep, false, null)]
+        public void ExpectedContextSet(string propagationMode, string integration, int? samplingPriority, bool shouldInject, string expectedContext)
+        {
+            Enum.TryParse(propagationMode, ignoreCase: true, out DbmPropagationLevel dbmPropagationLevel);
+            Enum.TryParse(integration, ignoreCase: true, out IntegrationId integrationId);
+
+            // capture command and parameter sent
+            string sql = null;
+            byte[] context = null;
+            var connectionMock = new Mock<IDbConnection>(MockBehavior.Strict);
+            var commandMock = new Mock<IDbCommand>();
+            var parameterMock = new Mock<IDbDataParameter>();
+            connectionMock.Setup(c => c.CreateCommand()).Returns(commandMock.Object);
+            commandMock.SetupSet(c => c.CommandText = It.IsAny<string>())
+                       .Callback<string>(value => sql = value);
+            commandMock.Setup(c => c.CreateParameter()).Returns(parameterMock.Object);
+            commandMock.SetupGet(c => c.Parameters).Returns(Mock.Of<IDataParameterCollection>());
+            parameterMock.SetupSet(p => p.Value = It.IsAny<byte[]>())
+                         .Callback<object>(value => context = (byte[])value);
+
+            var span = _v0Tracer.StartSpan("mysql.query", parent: SpanContext.None, serviceName: "pouet", traceId: (TraceId)0xCAFE, spanId: 0xBEEF);
+            span.SetTraceSamplingPriority((SamplingPriority)samplingPriority.Value);
+
+            DatabaseMonitoringPropagator.PropagateDataViaContext(_v0Tracer, dbmPropagationLevel, integrationId, connectionMock.Object, "pouet", new Scope(parent: null, span, scopeManager: null, finishOnClose: false), new SqlTags());
+
+            if (shouldInject)
+            {
+                sql.Should().StartWith("set context_info ");
+                BitConverter.ToString(context).Replace("-", string.Empty).Should().Be(expectedContext);
```

Comment:
Why the `Replace("-", string.Empty)`? This looks like code that was used for GUIDs.

Reply:
This is the best `byte[]` to `string` method I could find that was available all the way down to netcore21. But it inserts dashes between bytes, which I found ugly, so I added that 🤷  (and the linter forced me to replace `""` with `string.Empty`)

Reply:
This is fine for a test (`Replace()` will allocate a new string). Just FYI for product code, we have optimized helpers in `HexString`.